web/admin/src/SurveyResults.js

```diff
@@ -153,15 +154,18 @@ class SurveyResults extends Component {
       item.newResults.forEach(item => {
         const title = item.question
         let answer = ''
-        if (typeof item.answer === 'object' && !item.answer.length) {
-          answer = stringifyForCsv(item.answer)
-        } else if (typeof item.answer === 'object' && item.answer.length) {
-          answer = item.answer.map(answer =>
+        const origAnswer = item.schemaVersion === 2 ? JSON.parse(item.answer) : item.answer
```

Comment:
since this line appears twice, let's refactor:

```js
const origAnswer = getAnswer(item)
```
and
```js
const getAnswer = item => item.schemaVersion === 2 ? JSON.parse(item.answer) : item.answer
```

This will remove schema versioning logic from business logic, and put the schema logic in one place in case it has to be fixed or updated (with a version 3, for example) in the future.